.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,21 @@
+name: CodeQL
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
Should this be run on the PR's branch so we catch potential issues before code is merged in?

Reply:
It's configured to run on both the main branch and PRs against the main branch below. I believe it runs on the main branch to determine the baseline.